server/src/main/java/org/candlepin/async/tasks/UndoImportsJob.java

```diff
@@ -164,11 +167,12 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         log.info("Deleting all pools originating from manifests for owner/org: {}", displayName);
 
         List<Pool> pools = this.poolManager.listPoolsByOwner(owner).list();
-        for (Pool pool : pools) {
-            if (this.poolManager.isManaged(pool)) {
-                this.poolManager.deletePool(pool);
-            }
-        }
+        this.poolManager.deletePools(
+            pools.stream()
+                .filter(this.poolManager::isManaged)
+                .collect(Collectors.toList()),
+            new ArrayList<>()
```

Comment:
No need to pass another argument here, the deletePools method has an overloaded version which only accepts a single list of pools. Other than that, this looks good

Reply:
Done.